#165305711 Setting up test environment

Bakley opened the pull request:
### What does this PR do?
Creates an appropriate folder/file structure for React/Redux testing environment 

### What are the tasks to be completed?
- Setup redux test folder structure
- Install jest, enzyme and required dependencies

### How can this be manually tested?
After cloning the repository:
```
$ git checkout ch-setting-up-test-env-165305711
$ npm install
$ npm run test:watch
```
### Screenshots
<img width="633" alt="Screenshot 2019-05-28 at 11 11 27" src="https://user-images.githubusercontent.com/7009215/58461903-66d70f80-8139-11e9-9f5d-6197b82683ef.png">



#### What are the relevant pivotal tracker stories?
[#165305711](https://www.pivotaltracker.com/n/projects/2326460/stories/165305711)

Ogutu-Brian commented on __tests__/App.test.jsx:
```diff
@@ -0,0 +1,9 @@
+    const div = document.createElement('div');
+    ReactDOM.render(<App />, div);
+    ReactDOM.unmountComponentAtNode(div);
+});
```
Hello @Bakley ,
Please add a new  line at the end of the file.

Bakley replied:
Done that

Ogutu-Brian commented on package.json:
```diff
@@ -18,30 +20,35 @@
     "react": "^16.8.6",
     "react-dom": "^16.8.6",
-    "react-redux": "^7.0.3",
-    "redux": "^4.0.1",
```
Hello,
Redux packages have been removed from the dependencies in this file. Please fix that, it is essential for the app development.

Bakley replied:
I added those back

IssaIan commented on .gitignore:
```diff
@@ -61,3 +61,5 @@ typings/
 .next
 .DS_Store
 dist/
+package-lock.json
```
Remove the package-lock.json file. I have sent a guide on how to resolve conflicts on it in the channel. Check it out.

IssaIan replied:
And push it again.
cc @Bakley

Bakley committed:
165305711-ch(Test): Setting up testing enviromnet
 - setting up testing enviromnet
 - Write basic test for reducers
[finishes #165305711]

IssaIan reviewed (changes requested):
Remove the package-lock.json file. I have sent a guide on how to resolve conflicts on it in the channel. Check it out.

Bakley commented:
> Remove the package-lock.json file. I have sent a guide on how to resolve conflicts on it in the channel. Check it out.

Okay, seen your message. Removing it now!

Ogutu-Brian reviewed (approved):
This is a good job!

koechkevin commented on package.json:
```diff
@@ -4,7 +4,8 @@
   "main": "index.js",
   "scripts": {
-    "test": "jest --coverage",
+    "test": "jest --coverage --colors",
```
@Bakley good work. Now have this command as `"test": "npm run lint src/**/*.{js,jsx} *.js && jest --coverage --colors"` so that tests will fail when you have linting errors

Bakley replied:
Okay, thanks @koechkevin added the command

Philipotieno commented:
@Bakley This is a good job
